services/provider/server/server.go

```diff
@@ -969,6 +986,223 @@ func (s *OCSProviderServer) GetStorageClaimConfig(ctx context.Context, req *pb.S
+			desiredFlattenVRCName := fmt.Sprintf("rbd-flatten-volumereplicationclass-%v", util.FnvHash(volumeReplicationClass5mSchedule))
+			if vrc.Name == desiredVRCName || vrc.Name == desiredFlattenVRCName {
+				extR = append(extR,
+					&pb.ExternalResource{
+						Name:        vrc.Name,
+						Kind:        "VolumeReplicationClass",
+						Labels:      vrc.Labels,
+						Annotations: vrc.Annotations,
+						Data:        mustMarshal(vrc.Spec),
+					})
+			}
+
+		case snapapi.SchemeGroupVersion.WithKind("VolumeSnapshotClass"):
+			vsc := resource.(*snapapi.VolumeSnapshotClass)
+			vscName := util.GenerateNameForSnapshotClass(storageCluster.Name, util.RbdSnapshotter)
+			if vsc.Name == vscName {
+				extR = append(extR,
+					&pb.ExternalResource{
+						Name:        vsc.Name,
+						Kind:        "VolumeSnapshotClass",
+						Labels:      vsc.Labels,
+						Annotations: vsc.Annotations,
+						Data:        mustMarshal(vsc.Parameters),
+					})
+			}
+
+		case groupsnapapi.SchemeGroupVersion.WithKind("VolumeGroupSnapshotClass"):
+			vgsc := resource.(*groupsnapapi.VolumeGroupSnapshotClass)
+			var vgscName string
+			vgscName = util.GenerateNameForGroupSnapshotClass(storageCluster, util.RbdGroupSnapshotter)
+			if vgsc.Name == vgscName {
+				extR = append(extR,
+					&pb.ExternalResource{
+						Name:        vgsc.Name,
+						Kind:        "VolumeGroupSnapshotClass",
+						Labels:      vgsc.Labels,
+						Annotations: vgsc.Annotations,
+						Data:        mustMarshal(vgsc.Parameters),
+					})
+			}
+
+		case storagev1.SchemeGroupVersion.WithKind("StorageClass"):
+			storageClass := resource.(*storagev1.StorageClass)
+			if storageClass.Name == req.StorageClaimName {
+				extR = append(extR,
+					&pb.ExternalResource{
+						Name:        storageClass.Name,
+						Kind:        "StorageClass",
+						Labels:      storageClass.Labels,
+						Annotations: storageClass.Annotations,
+						Data:        mustMarshal(storageClass.Parameters),
+					})
+			}
+
+		case csiopv1a1.GroupVersion.WithKind("ClientProfile"):
+			clientProfile := resource.(*csiopv1a1.ClientProfile)
+			clientProfileName := consumerConfig.GetRbdClientProfileName()
+			if clientProfile.Name == clientProfileName {
+				extR = append(extR,
+					&pb.ExternalResource{
+						Name:        clientProfile.Name,
+						Kind:        "ClientProfile",
+						Labels:      clientProfile.Labels,
+						Annotations: clientProfile.Annotations,
+						Data:        mustMarshal(clientProfile.Spec),
+					})
+			}
+
+		}
+
+	}
+
+	klog.Infof("successfully returned the storage class claim %q for %q", req.StorageClaimName, req.StorageConsumerUUID)
+	return extR, nil
+
+}
+
+// SHIM for GetStorageClaimConfig to return response to 4.18 clients
+func (s *OCSProviderServer) proxyGetStorageClaimConfigCephFs(
+	req *pb.StorageClaimConfigRequest,
+	storageCluster *ocsv1.StorageCluster,
+	consumerConfig util.StorageConsumerResources,
+	kubeResources []client.Object,
```

Comment:
All of the comments raised for the `proxyGetStorageClaimConfigBlock` should also apply to this function and the implementation in its body